game/server/tf/bot/tf_bot.cpp

```diff
@@ -1274,12 +1274,29 @@ void CTFBot::ChangeTeam( int iTeamNum, bool bAutoTeam, bool bSilent, bool bAutoB
 //-----------------------------------------------------------------------------------------------------
 bool CTFBot::ShouldGib( const CTakeDamageInfo &info )
 {
-	// only gib giant/miniboss
-	if ( TFGameRules()->IsMannVsMachineMode() && ( IsMiniBoss() || GetModelScale() > 1.f ) )
-	{
-		return true;
+	if (CTFGameRules_IsPVEModeActive(*g_pGameRules) && CBaseEntity_GetTeamNumber(this) == TFTEAM_BLUE) {
```

Comment:
Why not `TFGameRules()->IsMannVsMachineMode()`?
Why `CBaseEntity_GetTeamNumber` instead of `GetTeamNumber()`?